bedboss/bedqc/bedqc.py

```diff
@@ -0,0 +1,105 @@
+import logging
+import os
+import tempfile
+import pypiper
+import subprocess
+
+from bedboss.const import MAX_FILE_SIZE, MAX_REGION_SIZE, MIN_REGION_WIDTH
+from bedboss.exceptions import QualityException
+
+
+_LOGGER = logging.getLogger("bedboss")
+
+
+def bedqc(
+    bedfile: str,
+    outfolder: str,
+    max_file_size: int = MAX_FILE_SIZE,
+    max_region_size: int = MAX_REGION_SIZE,
+    min_region_width: int = MIN_REGION_WIDTH,
+    pm: pypiper.PipelineManager = None,
+    **kwargs,
+) -> bool:
+    """
+    Main pipeline function
+    :param bedfile: path to the bed file
+    :param outfolder: path to the folder where to store information about pipeline and logs
+    :param max_file_size: maximum file size
+    :param max_region_size: maximum region size
+    :param min_region_width: min region width
+    :param pm: pypiper object
+    :return: True if file passed Quality check
+    """
+    _LOGGER.info("Running bedqc...")
+    _LOGGER.warning(f"Unused arguments: {kwargs}")
+
+    output_file = os.path.join(outfolder, "flagged_bed.csv")
+    bedfile_name = os.path.basename(bedfile)
+    input_extension = os.path.splitext(bedfile_name)[1]
+
+    file_exists = os.path.isfile(bedfile)
+
+    if not pm:
+        pm = pypiper.PipelineManager(
+            name="bedQC-pipeline", outfolder=outfolder, recover=True, multi=True
+        )
+
+    detail = []
+
+    # check number of regions
+    script_path = os.path.join(
+        os.path.dirname(os.path.dirname(os.path.abspath(__file__))),
+        "bedqc",
+        "est_line.sh",
+    )
+    assert os.path.exists(script_path), FileNotFoundError(
+        f"'{script_path}' script not found"
+    )
```

Comment:
is it worth finding a way to do this in python to avoid an external shell dependency? This is a very simple script and it should be possible to either write it using python functions, or at least just call the commands directly using subprocess. However, maybe that would eliminate the value added by pypiper? Just a thought, not sure if we need to do that.

Reply:
If I would write this code from the very beginning, I would probably did it using Python libraries, or just subprocess. It would simplify code a bit